### ISSUE TITLE
Missing plots from group.density past 1.50 version (not available in 1.52)

### ISSUE THREAD

**Issue by pcm32** [sneumann/xcms#184]
Hi there!

Thanks for the excellent work!

At version 1.50 and previously, the invocation of group.density was generating plots as per this line:

https://github.com/sneumann/xcms/blob/1f816d946ed079f95dc4efa4e0aaa7cca360ef64/R/methods-xcmsSet.R#L390

This behaviour is no longer available in current versions:

https://github.com/sneumann/xcms/blob/f11aac4020319d991b845154e492f1031f946888/R/do_groupChromPeaks-functions.R#L136

While I share the spirit of removing those plot executions, I wonder whether there is anywhere else in the code base some canned call that can be used to generate these same plots from the resulting xset produced by `group`, since we want to be able to fill that output file with what was currently there (plots previously generated). Otherwise, could you offer some guidance on quick snippet that would yield those results starting with the produced xset object? It doesn't seem obvious at least that all elements required for the plot are stored in the output object.

Thanks!
Pablo

**Comment by sneumann** [comment]
The plotting should be restored, to avoid breaking existing code. 
The question is where to do it. A choice could be to *not* do it in `do_groupChromPeaks_density()` 
for the sake of modularisation and keeping implicit side-effects (like plotting) to a minimum. 
In that case, it could be restored (or called from) https://github.com/sneumann/xcms/blob/f11aac4020319d991b845154e492f1031f946888/R/methods-xcmsSet.R#L343 
where we still have the `sleep` parameter that is controlling the plotting.

**Comment by jotsetung** [comment]
Was my fault. I removed the plotting functionality from the analysis algorithms because I wanted to separate the two things. I'll look into adding the visualizations again for the identified groups.

**Comment by pcm32** [comment]
Do you plan to make a new release to BioConductor when this is reinstated? To know roughly from where and when we could access the functionality. Thanks!

**Comment by jotsetung** [comment]
We will push the changes definitely to the Bioconductor devel branch - I'm not sure if we're going to add them also to the current Bioconductor release branch (i.e. Bioconductor 3.5).
Regarding timing I can't promise anything as I'll be on holidays soon - but should be done by Mid July - or earlier. I'll let you definitely know when it's done.

**Comment by jotsetung** [comment]
@pcm32 , the `sleep` parameter is working again in the `xcms` version 2.99.2 available in the current Bioconductor development version.
Alternatively, you can install it also from github:
```r
library(devtools)
install_github("sneumann/xcms")
```

sorry for the inconvenience.

**Comment by jotsetung** [comment]
@pcm32 , are you OK with the solution? can we close the issue?

**Comment by pcm32** [comment]
Thanks @jotsetung and @sneumann, apologies for not coming back. Yes, this creates the missing plots, but there are now some other plots that went away. Should I open a new issue for those? I won't be able to give more details in a few days I think though...

**Comment by jotsetung** [comment]
Would be nice to know which plots exactly you need. I did indeed remove most of the plots that are created *during* the calculations. On one hand to keep the plotting code separate from the calculation code and also (mainly) because these plots can be problematic if the code is run in parallel.

**Comment by pcm32** [comment]
I agree that is a good idea to get rid of those plots in those parts of the code. What I would suggest then would be to provide some code snippet or function call that would allow users to invoke the creation of those plots starting from some R object produced by those computations. I'll try to dig in soon on the code to see where are the missing plots.

**Comment by jotsetung** [comment]
I've re-added some more plotting functions. Closing this issue for now. Feel free to re-open if you still find some functions that are missing.

You can get the updated version (still devel) from github:
```r
library(devtools)
install_github("sneumann/xcms")
```
